internal/scorer/algorithm/wam/wam.go

```diff
@@ -23,26 +23,28 @@ import (
 
 const Name = "weighted_arithmetic_mean"
 
-type WeighetedArithmeticMean struct {
+// WeightedArithmeticMean is an implementation of the Weighted Arithmetic.
+// "Weighted Arithmetic Mean" is also known as "Weighted Average".
```

Comment:
The first sentence might need the work "Mean" at the end.

A blank line between each sentence would be clearer.

Also - a link to the Wikipedia page may be useful too!

Reply:
Thanks for the "Mean" catch!

Included a blank line and the Wikipedia page link.